adala/runtimes/_openai.py

```diff
@@ -42,16 +43,24 @@ async def async_create_completion(
     Returns:
         Dict[str, Any]: OpenAI response or error message.
     """
-    openai_api_key = openai_api_key or os.getenv("OPENAI_API_KEY")
-    messages = [{"role": "user", "content": user_prompt}]
+    openai_api_key = (
+        openai_api_key or client.api_key or os.environ['OPENAI_API_KEY']
+    )
+    messages = [{'role': 'user', 'content': user_prompt}]
     if system_prompt:
         if instruction_first:
             messages.insert(0, {"role": "system", "content": system_prompt})
         else:
             messages[0]["content"] += system_prompt
 
     try:
-        completion = await client.chat.completions.create(
+        # FIXME: I think we're better off doing this globally, since this
```

Comment:
I have some thoughts on how we could do this, and have tested an implementation, but def want to discuss more before pulling the trigger here.